File: pset6/credit/credit.py
from cs50 import get_int


def main():
    cardnumber = get_int("Number: ")
    # checking the length of card
    length = len(str(cardnumber))
    # if length is normal > check sum of card, else print "INVALID"
    if length == 13 or length == 15 or length == 16:
        # if sum is OK, give an answer
        if checksum(length, cardnumber):
            answer(cardnumber)
        else:
            print("INVALID")
    else:
        print("INVALID")


def checksum(length, cardnumber):
    sum = 0
    for i in range(length):
        # if digit is off (from last digit of cardnumber)
        if i % 2 == 0:
            sum += cardnumber % 10
        # elif number need'nt to plus digit to remain
        elif cardnumber % 10 < 5:
            sum += (cardnumber % 10) * 2
        else:
            sum += (((cardnumber % 10) * 2) % 10) + (((cardnumber % 10) * 2) // 10)
        cardnumber //= 10
    # I'm trying to use ternar conditional
    # total sum is correct (divides by 10)
    return True if sum % 10 == 0 else False


def answer(cardnumber):
    # I'll use additional var string to check 1'st and 2'nd symbol of cardnumber
    string = str(cardnumber)
    if (len(str(cardnumber)) == 13 or len(str(cardnumber))) == 16 and string[0] == '4':
        print("VISA")
    elif len(str(cardnumber)) == 16 and string[0] == '5' and string[1] in ['1', '2', '3', '4', '5']:
        print("MASTERCARD")
    elif len(str(cardnumber)) == 15 and string[0] == '3' and string[1] in ['4', '7']:
        print("AMEX")
    else:
        print("INVALID")
    return True


if __name__ == "__main__":
    main()

File: pset1/credit/credit.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    long long cardnumber;
    int length;
    bool checksum();
    int checklen();
    int answer();
    cardnumber =  get_long_long("Number:");
    length = checklen(cardnumber); //checking the length of card
    if (length == 13 || length == 15 || length == 16) //if length is normal > check sum of card, else print "INVALID"
    {
        if (checksum(length, cardnumber) == true) //if sum is OK, give an answer
        {
            answer(cardnumber);
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }
}

bool checksum(int length, long long cardnumber) //checks sum of card
{
    int sum = 0;
    for (int i = 0; i < length; ++i, cardnumber /= 10) //each iteration removes last number
    {
        if (i % 2 == 0) //if digit is odd (from last digit of cardnumber)
        {
            sum += (cardnumber % 10);
        }
        else if ((cardnumber % 10) < 5)// else if number need'nt to plus digit to remain
        {
            sum += ((cardnumber % 10) * 2);
        }
        else
        {
            sum += (((cardnumber % 10) * 2) % 10) + (((cardnumber % 10) * 2) / 10);
        }
    }
    if (sum % 10 == 0) // if total sum is correct (divides by 10)
    {
        return true;
    }
    else
    {
        return false;
    }
}

int checklen(long long cardnumber) //gives length of card number
{
    long long number = cardnumber;
    int len = 0;
    while (number > 0)
    {
        number /= 10;
        len++;
    }
    return len;
}

int answer(long long cardnumber) //prints card company
{
    //check 4 for visa
    if ((cardnumber >= 4000000000000 && cardnumber < 5000000000000) || (cardnumber >= 4000000000000000 &&
            cardnumber < 5000000000000000)) // ~~~~~~~~~~~~~~ ASK!!!!!!!! Why style50 needs more tabs ~~~~~~~~~~
    {
        printf("VISA\n");
    }
    //checks 51 52 53 54 55 for mastercard and 34 or 37 for amex
    else if (cardnumber >= 5100000000000000 && cardnumber < 5600000000000000)
    {
        printf("MASTERCARD\n");
    }
    //checks 34 or 37 for amex
    else if ((cardnumber >= 340000000000000 && cardnumber < 350000000000000) || (cardnumber >= 370000000000000 &&
             cardnumber < 380000000000000))
    {
        printf("AMEX\n");
    }
    else
    {
        printf("INVALID\n");
    }
    return 0;
}